src/SFML/Window/WindowBase.cpp

```diff
@@ -207,18 +211,67 @@ void WindowBase::setSize(const Vector2u& size)
 {
     if (m_impl)
     {
-        m_impl->setSize(size);
+        // Constrain requested size within minimum and maximum bounds
+        const auto minimumSize = m_impl->getMinimumSize().value_or(sf::Vector2u());
+        const auto maximumSize = m_impl->getMaximumSize().value_or(
+            sf::Vector2u(std::numeric_limits<unsigned int>::max(), std::numeric_limits<unsigned int>::max()));
+        const auto width  = std::clamp(size.x, minimumSize.x, maximumSize.x);
+        const auto height = std::clamp(size.y, minimumSize.y, maximumSize.y);
+
+        // Do nothing if requested size matches current size
+        const sf::Vector2u clampedSize(width, height);
+        if (clampedSize == m_size)
+            return;
+
+        m_impl->setSize(clampedSize);
 
         // Cache the new size
-        m_size.x = size.x;
-        m_size.y = size.y;
+        m_size = clampedSize;
 
         // Notify the derived class
         onResize();
     }
 }
 
 
+////////////////////////////////////////////////////////////
+void WindowBase::setMinimumSize(const std::optional<Vector2u>& minimumSize)
+{
+    if (m_impl)
+    {
+        [[maybe_unused]] const auto minimumSizeIsLessThanMaximumSize = [this, minimumSize]()
+        {
+            if (!minimumSize.has_value() || !m_impl->getMaximumSize().has_value())
+                return true;
+            return minimumSize->x < m_impl->getMaximumSize()->x && minimumSize->y < m_impl->getMaximumSize()->y;
+        };
+        assert(minimumSizeIsLessThanMaximumSize() && "Minimum size must be less than maximum size");
```

Comment:
I'm not so sure we should assert here. I think clamping and outputting a descript error would be better.

Output examples:
"Maximum size (200x400) is smaller than minimum size (300x200)! Ignoring"
"Minimum size (900x700) is larger than maximum size (800x600)! Ignoring"

Make sure this goes to stderr.

It would probably look something like this:
```c++
if (!minimumSizeIsLessThanMaximumSize())
{
    std::cerr << "Minimum size (" << minimumSize.x << "x" << minimumSize.y << ") is larger than maximum size (" << m_impl->getMaximumSize()->x << "x" << m_impl->getMaximumSize()->y << ")! Ignoring\n";
    return;
}
```

Reply:
Also, inverting the output and renaming the `minimumSizeIsLessThanMaximumSize` function to `minimumSizeIsGreaterThanMaximumSize` may be more optimal in this situation.

Reply:
Another suggestion would be to assert and do an early return that way debug builds crash and release builds have well defined behavior.

Reply:
When writing much of this I knew there were a lot of options for error detection. I thought using an `assert` was an easy place to start but I'm open to a solution like this. There are a number ways we can define what happens when users provide conflict minimum and maximum sizes. I'm curious to hear the rest of team weigh in on how we should handle this case.